dcrsqlite/apisource.go

```diff
@@ -320,12 +317,83 @@ func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes
 
 	msgTx := txhelpers.MsgTxFromHex(txhex)
 	if msgTx == nil {
-		return tx, prevOutAddresses, "", nil
+		return &apitypes.ExplorerTxData{
+			Tx:       tx,
+			VinAddrs: prevOutAddresses,
+		}
 	}
 
 	txType := txhelpers.DetermineTxTypeString(msgTx)
+	var txFee, txFeeRate float64
+	if txType == "Ticket" || (txType == "Regular" && len(tx.Vin[0].Coinbase) < 5) {
+		txFee = txhelpers.TxFee(msgTx)
+		txFeeRate = txhelpers.TxFeeRate(msgTx)
+	} else {
+		txFee = 0.0
+		txFeeRate = 0.0
+	}
+
+	vinfo := new(apitypes.VoteInfo)
```

Comment:
no need to do an actual alloc here since it's done below.  can just `var vinfo *apitypes.VoteInfo`